okx/exchain
# optimize group deliver txs' prepare

add the init

remove useless line

hide original function

add spin lock

add go routine pool

add depency

add log

to test

add debug log

use pointer

delete debug info

add sync outter

replace waitgroup with spinlock

fix spin lock

fix spinlock

reuset txmanager

fix the flag

reuse txReps and extraTxsInfo

add clear

fix equal situation

del useless lines



- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Updated relevant documentation (`docs/`)

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))
